ovs-dpdk.md

```diff
@@ -0,0 +1,180 @@
+
+Number of pages can vary according to usage requirements, but is limited to memory size.
+Hugepages must be marked as shared in order to enable host and guest communication, as packets are transmitted and received on buffers allocated on hugepages memory. 
+
+Please note that 1Gib hugepages feature is not supported on every CPU 
+
+Review custom properties
+```
+# engine-config -g UserDefinedVMProperties
+```
+Add engine custom properties for hugepages support (followed by engine restart)
+```
+# engine-config -s "UserDefinedVMProperties=hugepages=^.*$;hugepages_shared=(true|false)"
+```
+ 
+
+### Enable DPDK in OVS:
+
+ovs-vscwhitchd should get a few parameters in order to enable DPDK functionality.
+
+* Set ```dpdk-init``` so OVS would initialize and support DPDK ports:
+```
+# ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-init=true
+```
+ 
+
+* ```dpdk-socket-mem```: a comma-separated list of amount of memory to allocate from each NUMA node. It is important to allocate hugepage memory to all NUMA nodes that have DPDK interfaces associated with them, otherwise, the NUMA node cannot be used and errors will occur. 
+For example, in a dual socket system, we allocate 1GiB from NUMA node 0 and 0GiB from the second:
+
+```
+# ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-socket-mem="1024,0" 
+```
+
+* ```pmd-cpu-mask```: PMD (poll mode driver) threads are used to poll DPDK devices for new packets instead of using using the NIC driver that interacts with the kernel. 
+To better scale the workloads across cores, multiple pmd threads can be created and pinned to CPU cores by explicitly specifying pmd-cpu-mask. Ensure that cores from the same NUMA nodes as the network device are enabled. In case of Hyper Threading, it is best to use sibling cpus.
+
+eg: To spawn 2 pmd threads and pin them to cores 1, 2:
+
+```
+# ovs-vsctl set Open_vSwitch . other_config:pmd-cpu-mask=6
+
+```
+
+* ```dpdk-lcore-mask```: Specifies the CPU cores on which dpdk lcore threads should be spawned and expects hex string. These are threads for OVS usage other than PMD. They cannot be same as PMD or vCPUs threads.
+ 
+ Restart openvswitch service
+```
+# sudo systemctl restart openvswitch
+```
+ 
+Verify DPDK is enabled:
+```
+# ovs-vsctl get Open_vSwitch . iface_types
+[dpdk, dpdkr, dpdkvhostuser, dpdkvhostuserclient, geneve, gre, internal, ipsec_gre, lisp, patch, stt, system, tap, vxlan]
+```
+
+## OVS-DPDK in oVirt
+ 
+* Upon creating a new cluster, choose OVS as the switch type.
+* When setting up host networks, DPDK-bound NICs are shown in the dialog,as well as Linux-controlled ones
+* When creating a network on DPDK device, an OVS bridge with datapath_type=’netdev’ will be automatically attached to dpdk device will be created.
```

Comment:
Suggest:
"When creating a network on a DPDK device, an OVS bridge with datapath_type=’netdev’ automatically attached to a DPDK device, will be created."